### PR TITLE
Create Component

### PR DESCRIPTION
**Pull request by xerona**
This isolates the truncation logic when we don't need the additional features of smart-text that can be more cumbersome.

This may account for half of https://jira.inbcu.com/browse/AUG-688

### CODE REVIEW

File: src/truncate-text.component.scss [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    &.TruncateText-truncated {
+      cursor: pointer;
+      & + /deep/ supre-tooltip {
+        display: inline !important;
+      }
+    }
+  }
+}
+.TruncateText-clone {
```
Comment (craigmichaelmartin) [inline src/truncate-text.component.scss]: Can remove this.

Comment (xerona) [inline src/truncate-text.component.scss]: Thanks!

File: src/truncate-text.component.scss [hunk 2]
```diff
@@ -0,0 +1,26 @@
+:host {
+  .TruncateText {
+    display: -webkit-box;
+    -webkit-box-orient: vertical;
+    overflow: hidden;
+    text-overflow: ellipsis;
+
+    & + /deep/ supre-tooltip {
```
Comment (xerona) [inline src/truncate-text.component.scss]: This style forces it to stay hidden even on hover if the text is not truncated. The next style allows it to show when it has been truncated.

File: src/truncate-text.component.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    | 'auto left'
+    | 'auto right' = 'auto top';
+
+  private truncated = false;
+
+  constructor() {}
+
+  ngOnInit() {}
+
+  @HostListener('mouseover', ['$event.target'])
```
Comment (xerona) [inline src/truncate-text.component.ts]: This method checks to see if the text has been truncated so we can use a class to allow the tooltip to be visible.
The underlying reason here is we only show a tooltip if the text is truncated, but we don't know on render if the text will be truncated by any other event like resize so we address it with this event.

**Comment by xerona** [comment]
@craigmichaelmartin ^^

**Comment by craigmichaelmartin** [comment]
Yep, already on it :) https://github.com/CINBCUniversal/prettier-flow. Will ping everyone when its ready for review.

**Comment by paulkahle** [comment]
@xerona 
This looks good.  Simple and to the point.

**Comment by xerona** [comment]
@paulkahle - thanks! I don't know how to add you to the repo so i can add you as a reviewer

**Pull request merged**